### PR TITLE
Vox traders spawn with money, have an option to spawn with an ID

### PR DESCRIPTION
**Pull request by unid15**
- Vox traders' wallets now spawn with a random amount of money and a coin. Amount ranges from $1 to $2000.
- When vox traders spawn, they have an option to start with a traveler's ID. It has no access or virtual wallet, and is used purely for identification.
- Lore-wise, traveler's ID are required to legally travel in human-controlled territories. They're compatible with the station's systems, so it's possible to add access to them.

The option to start without an ID is included for roleplayers and antagonists.

### CODE REVIEW

**Comment by ghost** [comment]
Update spesslaw with details on this?

**Comment by carshalash** [comment]
Why do the bird id's not have any access?

**Comment by Intigracy** [comment]
Why would they have access to the station if they aren't part of it?

**Comment by ihadtoregisterforthis** [comment]
Create vox outpost access and give it exclusively to them.
But IDs would be obligatory.

**Comment by 9600bauds** [comment]
Fan of these changes, but I don't see the point of the 30 second limit in picking an ID
In any case the default behavior should be to spawn one, I mean if you accidentally get one without meaning to you can just... throw it away?

**Comment by DeityLink** [comment]
ID cards should always have a virtual wallet, even if empty, or you will run into runtime errors. Keep that in mind.

**Comment by Brolaire** [comment]
Amazing, all I ever asked for

**Comment by Probe1** [comment]
Seems ok but it feels like the ID thing will lead to "Why don't all jobs have it"

**Comment by DeityLink** [comment]
The Vox should have a machine that lets them print IDs, or re-brand ID names, while being stuck with the default ID icon, a few restricted job titles, and no ability to set up accesses on them. Then you'd see people (traitors) come to the Vox outpost to request a name change on their ID (or just break their way in to use the machine).

Just a random idea like that.

**Comment by Probe1** [comment]
I like the thought but there's a lot of similar thoughts tied to "a machine" that would be a stand in for an npc you would order items from to sell.  Pay 50 credits get something sell it for 75.

**Comment by DeityLink** [comment]
The machine itself would be free to use, I was implying that the Vox would force visitors to pay to use their machine. That's how lots of commerces work IRL you know?

File: code/game/jobs/job/whitelisted.dm [hunk 1]
```diff
@@ -38,9 +41,27 @@
...
+			if(!job_master)
+				to_chat(H, "Job controller doesn't exist. Something is wrong!")
+				return
+			if((current_time + 30 SECONDS) <= world.time)
+				to_chat(H, "<span class='danger'>Too slow, sorry.</span>")
+				return
+
+			no_id = 0
+			job_master.spawnId(H, "Trader", "Traveler")
+			to_chat(H, "<span class='info'>Ah, there's my ID. I was almost worried for a second that I had lost it.</span>")
```
Comment (Zth--) [inline code/game/jobs/job/whitelisted.dm:64]: amazing

**Comment by NigglyWiggly** [comment]
@DeityLink 

> Vox should have a machine that lets them play with their IDs 

But why
Why on earth would that be a good idea or even a necessary one

File: code/game/jobs/job/whitelisted.dm (hunk 1 above)
Comment (NigglyWiggly) [inline code/game/jobs/job/whitelisted.dm:64]: But why

**Comment by DeityLink** [comment]
Any things exclusive to the Vox outpost that we can come up with are good ideas. Currently playing Vox trader is generally summarized as "find a way to get to the station, then basically be an assistant"

**Comment by 9600bauds** [comment]
I still don't understand why is there even a 30 second limit on picking this

**Pull request merged**